src/toil/common.py

```diff
@@ -498,20 +568,116 @@ def parser_with_common_options(
+            workDir = values
+            if workDir is not None:
+                workDir = os.path.abspath(workDir)
+                if not os.path.exists(workDir):
+                    raise RuntimeError(f"The path provided to --workDir ({workDir}) does not exist.")
+
+                if len(workDir) > 80:
+                    logger.warning(f'Length of workDir path "{workDir}" is {len(workDir)} characters.  '
+                                   f'Consider setting a shorter path with --workPath or setting TMPDIR to something '
+                                   f'like "/tmp" to avoid overly long paths.')
+            setattr(namespace, self.dest, values)
+    return CheckWorkDir
+
+
+def make_int_range_validation_action(min: int, max: Optional[int]=None) -> Type[_StoreAction]:
+    class RangeValidation(_StoreAction):
+        def __call__(self, parser: Any, namespace: Any, values: Any, option_string: Any=None) -> None:
+            try:
+                if not iC(min, max)(values):
+                    raise parser.error(f"The {option_string} option is out of range: {values}")
+            except AssertionError:
+                raise RuntimeError(f"The {option_string} option has an invalid value: {values}")
+            setattr(namespace, self.dest, values)
+    return RangeValidation
+
+
+def make_float_range_validation_action(min: float, max: Optional[float]=None) -> Type[_StoreAction]:
+
+    class RangeValidation(_StoreAction):
+        def __call__(self, parser: Any, namespace: Any, values: Any, option_string: Any=None) -> None:
+            try:
+                if not fC(min, max)(values):
+                    raise parser.error(f"The {option_string} option is out of range: {values}")
+            except AssertionError:
+                raise RuntimeError(f"The {option_string} option has an invalid value: {values}")
+            setattr(namespace, self.dest, values)
+    return RangeValidation
+
+
+def make_closed_interval_check_action(min: Union[int, float], max: Optional[Union[int, float]]=None) -> Type[_StoreAction]:
+    class RangeValidation(_StoreAction):
+        def __call__(self, parser: Any, namespace: Any, values: Any, option_string: Any=None) -> None:
+            def is_within(x: Union[int, float]) -> bool:
+                if max is None:
+                    return min <= x
+                else:
+                    return min <= x <= max
+            try:
+                if not is_within(values):
+                    raise parser.error(f"{option_string} ({values}) must be within the range: [{min}, {'infinity' if max is None else max}]")
+            except AssertionError:
+                raise RuntimeError(f"The {option_string} option has an invalid value: {values}")
+            setattr(namespace, self.dest, values)
+    return RangeValidation
+
+
+def make_check_ssekey_action() -> Type[_StoreAction]:
+    class SSEKeyValidation(_StoreAction):
```

Comment:
I feel like the action names could be better. Some of these are imperatives ("Check whatever"), some are nouns for processes ("Whatever validation"). And in default argparse you say things like `action="store_true"` where the name is a noun for the action that is going to happen.

We should settle on one naming scheme, preferably one that argparse already has some evidence of using for its actions, and we should use it consistently for all our custom actions. Maybe this would be `StoreSSEKey`? Or `ParseSSEKey`? Or, since Argparse ships a `BooleanOptionalAction`, maybe this is an `SSEKeyAction`? Or `ValidatedSSEKeyAction`?